l1-contracts/contracts/bridge/L1Nullifier.sol

```diff
@@ -741,4 +741,37 @@ contract L1Nullifier is IL1Nullifier, ReentrancyGuard, Ownable2StepUpgradeable,
     function unpause() external onlyOwner {
         _unpause();
     }
+
+    /*//////////////////////////////////////////////////////////////
+                            LEGACY INTERFACE
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Legacy function to finalize withdrawal via the same
+    /// interface as the old L1SharedBridge.
+    /// @dev Note, that we need to keep this interface, since the `L2AssetRouter`
+    /// will continue returning the previous address as the `l1SharedBridge`. The value
+    /// returned by it is used in the SDK for finalizing withdrawals.
+    /// @param _chainId The chain ID of the transaction to check
+    /// @param _l2BatchNumber The L2 batch number where the withdrawal was processed
+    /// @param _l2MessageIndex The position in the L2 logs Merkle tree of the l2Log that was sent with the message
+    /// @param _l2TxNumberInBatch The L2 transaction number in the batch, in which the log was sent
+    /// @param _message The L2 withdraw data, stored in an L2 -> L1 message
+    /// @param _merkleProof The Merkle proof of the inclusion L2 -> L1 message about withdrawal initialization
+    function finalizeWithdrawal(
```

Comment:
This does a L1N -> L1AR -> L1N routing. Maybe it makes sense to call the finalizeDeposit directly here? But it is also ok like this, it might overall be less complexity

Reply:
agree